src/option_menu.c

```diff
@@ -0,0 +1,929 @@
+        break;
+    case 2:
+        sub_80886D4();
+        break;
+    case 3:
+        if (sub_80886F0() != TRUE)    //silly matching quirk
+            return;
+        break;
+    case 4:
+        sub_8088BD0();
+        break;
+    case 5:
+        sub_8088C0C();
+        break;
+    case 6:
+        sub_8088D8C();
+        break;
+    case 7:
+        for (i = 0; i < MENUITEM_COUNT; i++)
+            sub_80889A8(i);
+        break;
+    case 8:
+        sub_8088DE0(sOptionsMenuPtr->unkE);
+        break;
+    case 9:
+        sub_8088680();
+        break;
+    default:
+        sub_8088514();
+    }
+    sOptionsMenuPtr->state++;
+}
+
+//sub_8088514
+static void sub_8088514(void)
+{
+    CreateTask(sub_8088780, 0);
+    SetMainCallback2(sub_808835C);
+}
+
+
+//sub_8088530
+static void sub_8088530(void)
+{
+    void * dest = (void *) VRAM;
+    DmaClearLarge16(3, dest, VRAM_SIZE, 0x1000);
+    
+    DmaClear32(3, (void *)OAM, OAM_SIZE);
+    DmaClear16(3, (void *)PLTT, PLTT_SIZE);    
+    SetGpuReg(REG_OFFSET_DISPCNT, 0);
+    ResetBgsAndClearDma3BusyFlags(0);
+    InitBgsFromTemplates(0, sOptionMenuBgTemplates, ARRAY_COUNT(sOptionMenuBgTemplates));
+    ChangeBgX(0, 0, 0);
+    ChangeBgY(0, 0, 0);
+    ChangeBgX(1, 0, 0);
+    ChangeBgY(1, 0, 0);
+    ChangeBgX(2, 0, 0);
+    ChangeBgY(2, 0, 0);
+    ChangeBgX(3, 0, 0);
+    ChangeBgY(3, 0, 0);
+    InitWindows(sOptionMenuWinTemplates);
+    DeactivateAllTextPrinters();
+    SetGpuReg(REG_OFFSET_BLDCNT, 0xC1);
+    SetGpuReg(REG_OFFSET_BLDY, 2);
+    SetGpuReg(REG_OFFSET_WININ, 1);
+    SetGpuReg(REG_OFFSET_WINOUT, 0x27);
+    SetGpuReg(0, 0x3040);
+    ShowBg(0);
+    ShowBg(1);
+    ShowBg(2);
+};
+
+//sub_8088680
+static void sub_8088680(void)
+{
+    s32 x;
+    x = 0xE4 - GetStringWidth(0, gText_PickSwitchCancel, 0);
+    FillWindowPixelBuffer(2, 0xFF); 
+    AddTextPrinterParameterized3(2, 0, x, 0, sOptionsMenuPickSwitchCancelTextColor, 0, gText_PickSwitchCancel);
+    PutWindowTilemap(2);
+    CopyWindowToVram(2, 3);
+}
+
+
+//sub_80886D4
+static void sub_80886D4(void)
+{
+    ResetSpriteData();
+    ResetPaletteFade();
+    FreeAllSpritePalettes();
+    ResetTasks();
+    ScanlineEffect_Stop();
+}
+
+//sub_80886F0
+static bool8 sub_80886F0(void)
+{
+    switch (sOptionsMenuPtr->state2)
+    {
+    case 0:
+        LoadBgTiles(1, GetUserFrameGraphicsInfo(sOptionsMenuPtr->option[MENUITEM_FRAMETYPE])->tiles, 0x120, 0x1AA);
+        break;
+    case 1:
+        LoadPalette(GetUserFrameGraphicsInfo(sOptionsMenuPtr->option[MENUITEM_FRAMETYPE])->palette, 0x20, 0x20);
+        break;
+    case 2:
+        LoadPalette(sOptionsMenuPalette, 0x10, 0x20);
+        LoadPalette(stdpal_get(2), 0xF0, 0x20);
+        break;
+    case 3:
+        sub_814FDA0(1, 0x1B3, 0x30);
+        break;
+    default:
+        return TRUE;
+    }
+    sOptionsMenuPtr->state2++;
+    return FALSE;
+}
+
+
+
+// I could not get this function to match. GOTO statements weren't even compiling correctly.
```

Comment:
agbcc can tail merge cases. Try to write the code after goto respectively in those affected cases